File: Editorial/347_Top_K_Frequent_Elements/CPP/2_Quickselect.cpp
int partition(int left, int right, int pivot_index) {
    int pivot_frequency = count_map[unique[pivot_index]];
    // 1. Move the pivot to the end
    swap(unique[pivot_index], unique[right]);

    // 2. Move all less frequent elements to the left
    int store_index = left;
    for (int i = left; i <= right; i++) {
        if (count_map[unique[i]] < pivot_frequency) {
            swap(unique[store_index], unique[i]);
            store_index += 1;
        }
    }

    // 3. Move the pivot to its final place
    swap(unique[right], unique[store_index]);

    return store_index;
}

File: Editorial/0347_Top_K_Frequent_Elements/Java/2_Quickselect.java
public int partition(int left, int right, int pivot_index) {
    int pivot_frequency = count.get(unique[pivot_index]);
    // 1. Move the pivot to the end
    swap(pivot_index, right);
    int store_index = left;

    // 2. Move all less frequent elements to the left
    for (int i = left; i <= right; i++) {
        if (count.get(unique[i]) < pivot_frequency) {
            swap(store_index, i);
            store_index++;
        }
    }

    // 3. Move the pivot to its final place
    swap(store_index, right);

    return store_index;
}
